src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -17,102 +18,208 @@ internal class StringUtilities
         [MethodImpl(MethodImplOptions.AggressiveOptimization)]
         public static unsafe bool TryGetAsciiString(byte* input, char* output, int count)
         {
-            // Calculate end position
             var end = input + count;
-            // Start as valid
-            var isValid = true;
 
-            do
+            if (Sse2.IsSupported)
             {
-                // If Vector not-accelerated or remaining less than vector size
-                if (!Vector.IsHardwareAccelerated || input > end - Vector<sbyte>.Count)
+                if (Avx2.IsSupported && input <= end - Vector256<sbyte>.Count)
```

Comment:
I did some checks with higher threshold for AVX2 kicking in, but with not so good results. Hence AVX2 kicks in as soon there are enough elements.